File: Midtermlab3/MP15 – Export First 50 Rows to New CSV/MP15_exportRows.js
// MP15 - Export first 50 rows to new CSV file

const fs = require("fs");
const readline = require("readline");

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

rl.question("Enter CSV file path: ", function(inputPath) {

  rl.question("Enter output file name: ", function(outputPath) {

    try {
      const data = fs.readFileSync(inputPath, "utf8");
      const rows = data.split("\n");

      const first50 = rows.slice(0, 50).join("\n");

      fs.writeFileSync(outputPath, first50);

      console.log("\nFirst 50 rows exported successfully.");

    } catch (error) {
      console.log("Error:", error.message);
    }

    rl.close();
  });

});

File: Midtermlab3/MP15 – Export First 50 Rows to New CSV/MP15_ExportRows.java
// MP15 - Export first 50 rows of CSV to new file

import java.io.*;
import java.util.*;

public class MP15_ExportRows {

    public static void main(String[] args) {

        Scanner scanner = new Scanner(System.in);
        try {
            System.out.print("Enter CSV file path: ");
            String inputPath = scanner.nextLine();

            System.out.print("Enter output CSV file name: ");
            String outputPath = scanner.nextLine();

            try {
                BufferedReader br = new BufferedReader(new FileReader(inputPath));
                BufferedWriter bw = new BufferedWriter(new FileWriter(outputPath));

                String line;
                int count = 0;

                while ((line = br.readLine()) != null && count < 50) {
                    bw.write(line);
                    bw.newLine();
                    count++;
                }

                br.close();
                bw.close();

                System.out.println("\nFirst 50 rows exported successfully.");

            } catch (Exception e) {
                System.out.println("Error: " + e.getMessage());
            }

            System.out.println("Program Finished.");
        } finally {
            scanner.close();
        }
    }
}